docs/troubleshooting-guide/sql-logic.md

```diff
@@ -0,0 +1,261 @@
+    WITH date_series AS (
+    GENERATE_SERIES(...)
+    )
+
+    SELECT 
+    d.date
+    , SUM(p.price) AS total_sales
+    FROM date_series d
+    LEFT JOIN orders o
+        ON d.date = o.order_date
+    LEFT JOIN products p
+        ON o.product_id = p.id
+    GROUP BY 1
+    ;
+    ```
+6. If you’re getting an error message, go to [Troubleshooting error messages][troubleshooting-error-messages].
+7. If you're still stuck, search or ask the [Metabase community][discourse].
+
+#### How joins filter out unmatched rows
+| Join type | Rows that are filtered out |
+| --- | --- |
+| Table A INNER JOIN Table B | You will lose rows from both table A and table B if the join condition is not met. |
+| Table A LEFT JOIN Table B | You will lose rows from table B if the join condition is not met. |
+| Table B LEFT JOIN Table A | You will lose rows from table A if the join condition is not met. |
+| Table A OUTER JOIN Table B | You will lose rows from both table A and table B if the join condition is met. |
+| Table A FULL JOIN Table B | This join does not remove any rows from table A or table B. |
+
+**Explanation**
+
+Rows can get removed by accident when data gets refreshed in upstream systems or ETL jobs. 
+
+The order of the tables used in your `JOIN` clause matters. For example:
+
+- When you write a `LEFT JOIN`, the table that comes *before* the `JOIN` clause in your query is “on the left".
+- The rows from the table “on the right” will be filtered out if they don’t meet your join condition in the `ON` clause.
+
+The execution order of the query may combine your join conditions and WHERE clauses in ways that you might not expect.
+
+**Further reading**
+
+- [Common problems with SQL joins][common-join-problems] 
+
+#### My aggregations (counts, sums, etc.) are too high.
+
+1. Check if your source tables or upstream queries have [duplicated rows](#my-result-has-duplicated-rows).
+2. Check your source tables or upstream queries for rows that should be filtered out.
+    - Are you including empty or `NULL` rows in your aggregations?
+    - Ask your Metabase admin or data team about business logic that defines invalid, cancelled, or expired records that should be filtered out in queries.
+3. If you’re aggregating unique values, check that you’re not double-counting them.
+    - Do you need to use `COUNT_DISTINCT` instead of `COUNT`?
+    - Are you applying a `SUM` on top of a `COUNT_DISTINCT`?
+4. If you’re getting an error message, go to [Troubleshooting error messages][troubleshooting-error-messages].
+5. If you’re still stuck, search or ask the [Metabase community][discourse].
+
+#### My aggregations (counts, sums, etc.) are too low.
+
+1. Check if your source tables or upstream queries have [missing rows](#my-result-has-missing-rows).
+2. If you’re getting an error message, go to [Troubleshooting error messages][troubleshooting-error-messages].
+3. If you’re still stuck, search or ask the [Metabase community][discourse].
```

Comment:
Do we need each of these community links? Or just one "still stuck?" section at the bottom of the page (or pages if you split this page up).

Reply:
Yeah I think splitting the sections into new pages is a good idea. In general, I try to put an "exit" point at the end of each set of steps (on a long page) in case the reader doesn't get to the end!